internal/collector/sysinfo/platform/platform_linux.go

```diff
@@ -0,0 +1,221 @@
+package platform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/ubuntu/decorate"
+	"github.com/ubuntu/ubuntu-insights/internal/cmdutils"
+	"github.com/ubuntu/ubuntu-insights/internal/fileutils"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+// Info contains platform information for Linux.
+type Info struct {
+	WSL         WSL  `json:"wsl,omitzero"`
+	ProAttached bool `json:"proAttached,omitempty"`
+}
+
+// WSL contains platform information specific to Windows Subsystem for Linux.
+type WSL struct {
+	WSL              uint8  `json:"wsl,omitzero"`
+	WSLInterop       string `json:"wslInterop,omitempty"`
+	WSLVersion       string `json:"wslVersion,omitempty"`
+	WSLKernelVersion string `json:"wslKernelVersion,omitempty"`
+}
+
+type platformOptions struct {
+	root          string
+	detectVirtCmd []string
+	wslVersionCmd []string
+	proStatusCmd  []string
+}
+
+// defaultOptions returns options for when running under a normal environment.
+func defaultPlatformOptions() platformOptions {
+	return platformOptions{
+		root:          "/",
+		detectVirtCmd: []string{"systemd-detect-virt"},
+		wslVersionCmd: []string{"wsl.exe", "-v"},
+		proStatusCmd:  []string{"pro", "api", "u.pro.status.is_attached.v1"},
+	}
+}
+
+func (p Collector) collectPlatform() (info Info, err error) {
+	defer func() {
+		decorate.OnError(&err, "failed to collect platform information")
+	}()
+	info.WSL = p.collectWSL()
+	info.ProAttached = p.isProAttached()
+
+	return info, nil
+}
+
+// isWSL returns true if the system is running under Windows Subsystem for Linux.
+// This is done by checking the output of systemd-detect-virt.
+func (p Collector) isWSL() bool {
+	stdout, stderr, err := cmdutils.RunWithTimeout(context.Background(), 15*time.Second, p.platform.detectVirtCmd[0], p.platform.detectVirtCmd[1:]...)
+	if err != nil {
+		p.log.Warn("failed to run systemd-detect-virt", "error", err)
+		return false
+	}
+	if stderr.Len() > 0 {
+		p.log.Info("systemd-detect-virt output to stderr", "stderr", stderr)
+	}
+
+	if regexp.MustCompile(`wsl`).Match(stdout.Bytes()) {
```

Comment:
regex seems a bit overkill for a substring. Perhaps [Contains](https://pkg.go.dev/bytes#Contains) would be more appropriate?